2025-01-28 11:49:30.448 | INFO     | utils_logger:<module>:41 - Logging to file: logs\project_log.log
2025-01-28 11:49:30.448 | INFO     | __main__:main:93 - Starting CSV fetch demonstration...
2025-01-28 11:49:30.448 | INFO     | __main__:fetch_csv_file:52 - Fetching CSV data from https://raw.githubusercontent.com/MainakRepositor/Datasets/master/World%20Happiness%20Data/2020.csv...
2025-01-28 11:49:31.059 | INFO     | __main__:write_csv_file:76 - Writing CSV data to example_data\2020_happiness.csv...
2025-01-28 11:49:31.063 | INFO     | __main__:write_csv_file:80 - SUCCESS: CSV data written to example_data\2020_happiness.csv
2025-01-28 11:49:31.065 | INFO     | __main__:fetch_csv_file:56 - SUCCESS: CSV file fetched and saved as 2020_happiness.csv
2025-01-28 11:51:21.837 | INFO     | utils_logger:<module>:41 - Logging to file: logs\project_log.log
2025-01-28 11:51:21.837 | INFO     | __main__:main:93 - Starting Excel fetch demonstration...
2025-01-28 11:51:21.837 | INFO     | __main__:fetch_excel_file:52 - Fetching Excel data from https://raw.githubusercontent.com/denisecase/datafun-03-analytics/main/hosted/Feedback.xlsx...
2025-01-28 11:51:22.463 | INFO     | __main__:write_excel_file:76 - Writing Excel data to example_data\feedback.xlsx...
2025-01-28 11:51:22.467 | INFO     | __main__:write_excel_file:80 - SUCCESS: Excel data written to example_data\feedback.xlsx
2025-01-28 11:51:22.469 | INFO     | __main__:fetch_excel_file:56 - SUCCESS: Excel file fetched and saved as feedback.xlsx
2025-01-28 11:52:01.207 | INFO     | utils_logger:<module>:41 - Logging to file: logs\project_log.log
2025-01-28 11:52:01.209 | INFO     | __main__:main:94 - Starting JSON fetch demonstration...
2025-01-28 11:52:01.209 | INFO     | __main__:fetch_json_file:53 - Fetching JSON data from http://api.open-notify.org/astros.json...
2025-01-28 11:52:22.257 | ERROR    | __main__:fetch_json_file:61 - Request error occurred: HTTPConnectionPool(host='api.open-notify.org', port=80): Max retries exceeded with url: /astros.json (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x000001D3E0F6D6A0>, 'Connection to api.open-notify.org timed out. (connect timeout=None)'))
2025-01-28 11:53:55.730 | INFO     | utils_logger:<module>:41 - Logging to file: logs\project_log.log
2025-01-28 11:53:55.730 | INFO     | __main__:main:93 - Starting text fetch demonstration...
2025-01-28 11:53:55.730 | INFO     | __main__:fetch_txt_file:52 - Fetching text data from https://raw.githubusercontent.com/denisecase/datafun-03-analytics/main/hosted/romeo.txt...
2025-01-28 11:53:56.193 | INFO     | __main__:write_txt_file:76 - Writing data to example_data\romeo.txt...
2025-01-28 11:53:56.204 | INFO     | __main__:write_txt_file:80 - SUCCESS: Data written to example_data\romeo.txt
2025-01-28 11:53:56.205 | INFO     | __main__:fetch_txt_file:56 - SUCCESS: Text file fetched and saved as romeo.txt
